### ISSUE TITLE
Decorators for stable/unstable/etc. may not be the best idea.

### ISSUE THREAD

**Issue by felixchenier** [felixchenier/kineticstoolkit#44]
At the moment, the decorators @stable, @unstable, @private, @deprecated, @dead and @experimental are a nice way to automatically label functions that should or shouldn't be used. However, it creates deep stacks on errors, and sometimes the error messages or warnings are less clear than they should. For example: "This function has been deprecated". Which function? "This function is for private use only". Which function?

I'm starting to think that I over-engineered this documentation-only feature. Using a good convention may ultimately be more straightforward.

@stable --> Nothing
@unstable --> Precede with an underscore and document that this may become public someday using a standard convention.
@private --> Precede with an underscore, and do not document more since this is the standard for private function.
@deprecated --> Use a standard FutureWarning string, but that is customized to the feature that is being deprecated.
@dead --> Just remove from code.
@experimental --> Document it accordingly using a custom string.

I'm not set yet but I take these notes to not forget.

**Comment by felixchenier** [comment]
For transitions from unstable to stable, I would just have to code-refactor _function_name to function_name everywhere, but keep _function_name as a wrapper to function_name with a warning to use the new function from now on. This implies code refactoring where the old decorator method didn't, but it may add clarity to the code.

**Comment by felixchenier** [comment]
I left @unstable, @deprecated and @dead. Clearly these functions are not production functions and the decorators advantages beat the overhead and stack "pollution" for these functions.

@private was changed to the standard _ convention.
@stable was removed as it's now the default.
@experimental was replaced by a standard note in the docstring. No need to pollute the code.

I also added parameters to @deprecated and @dead so that we know since when the function is deprecated, until when, and what to do to workaround this deprecation.

This is resolved in commit bcbdfe4d956bab1008668b3511454e52cc552eba
